### PR TITLE
feat(grow): move intersections before interleave, add intersection-aware edges

### LINKED ISSUE #1124: feat(grow): move intersection detection before interleave phase (architectural)
### Problem

The current GROW phase order is:
```
serialize beats → interleave (predecessor edges) → gap beats → scene types → ... → path arcs → intersections
```

`interleave_beats` creates `predecessor` edges encoding temporal ordering between beats from different dilemmas. These edges create a **conditional prerequisite problem** for intersection detection: a beat that has a predecessor edge pointing to a beat on a disjoint path cannot be part of any intersection, because after intersection marking the beat would be shared across paths that don't have its prerequisite.

This makes intersection detection dependent on which beats the interleave phase "contaminated" with cross-path prerequisites — a fragile structural coupling that causes runtime failures (all intersection proposals rejected) and requires prompt engineering workarounds (prefer `_beat_01`, pre-filtering candidates).

The `phase3_incompatible_intersection` warnings in successful runs are evidence of this: even with the workarounds, some proposals are still rejected and discarded.

### Proposed Architectural Fix

Move intersection detection **before** `interleave_beats`:

```
serialize beats → intersections → resolve_temporal_hints → interleave → gap beats → scene types → ...
```

**Why this works**: Beats at this point have `belongs_to` edges (path membership), location and entity data, and narrative content — all the signals needed for intersection detection. They have no `predecessor` edges yet. Every beat is structurally eligible as an intersection candidate. The conditional prerequisite problem cannot occur on a clean beat DAG.

**Additional benefit — intersection-aware interleaving**: Once intersections are marked, the `interleave_beats` phase knows which beats are shared across paths. It can avoid creating predecessor edges that would break intersection compatibility. This makes interleaving structurally smarter, not just intersection-detection more reliable.

### Story Quality Impact Analysis

**Concern**: Intersections currently run after `path_arcs` (phase 4e), which assigns narrative arc roles (introduce, develop, commit, consequence) to beats. Moving intersections before path arcs means the model selects intersections without arc-position context.

**Assessment — small impact**:
- Intersection selection is primarily location/entity-based (beats share a location or a key character). This signal is available from beat serialization.
- Arc position is secondary context that enriches selection, not its primary basis.
- The loss of arc-position context is outweighed by a large gain: many more beats are eligible for intersection (no conditional prerequisite filtering), meaning the model can find more and better scene overlaps.

**Concern**: Gap beats are added after interleaving. If intersections run before gap beats, gap beats cannot be intersections.

**Assessment — acceptable**: Gap beats are path-local by construction (their predecessors exist on a single path only). They are already excluded from intersection candidates by the current code (`is_gap_beat` filter in `build_intersection_candidates`). This does not change.

**Concern**: Scene types and narrative enrichment phases (4a–4f) run after interleave. Moving intersections before these means intersection beats don't have scene type assignments.

**Assessment — small impact**: Scene types describe the dramatic function of a beat (action, revelation, confrontation). This context is useful but not required for intersection detection. The beat's `summary`, `location`, and `entities` fields already convey dramatic function informally.

**Net story quality impact**: Neutral to positive. More intersection candidates → more cross-dilemma scene overlap → richer branching narrative. The slight loss of arc/scene-type context for selection is a worthwhile trade.

### Design Doc Changes Required

- `docs/design/procedures/grow.md`: Move intersection phase in the phase sequence. Update phase dependency graph. Add note on intersection-aware interleaving.
- `docs/design/00-spec.md`: Update phase order diagram if present.

### Acceptance Criteria

- [ ] `intersections` phase runs before `interleave_beats` in the GROW phase graph
- [ ] `interleave_beats` receives intersection membership as context (or at minimum does not create edges that break existing intersections)
- [ ] `phase3_incompatible_intersection` warnings no longer appear (no conditional prerequisites on clean DAG)
- [ ] Intersection count in a standard run is equal or higher than current
- [ ] Design docs updated to reflect new phase order
- [ ] All existing GROW tests pass

### Verification

```bash
# No conditional prerequisite failures
grep 'phase3_incompatible_intersection' projects/*/logs/debug.jsonl
# should return 0 matches

# Intersections appear in graph before interleave edges
# (verify via snapshot diff between pre-intersection and post-intersection snapshots)
```

### PR DESCRIPTION
**Pull request by pvliesdonk**
## Summary

- Moves the `intersections` phase before `interleave_beats` in the GROW execution order (#1124)
- Running intersections on a clean beat DAG (no predecessor edges) eliminates the conditional-prerequisite rejection problem — the `check_intersection_compatibility` call always passes
- Adds intersection-awareness to `interleave_cross_path_beats`: skips predecessor edges between beats in the same intersection group (`interleave_skipped_same_intersection` log event)
- Updates all downstream phase priorities (+1 shift for scene_types through path_arcs)
- Updates design doc (`grow.md`) with new execution order note

**New execution order**: `validate_dag → intersections → interleave_beats → scene_types → narrative_gaps → pacing_gaps → atmospheric → path_arcs → entity_arcs → enumerate_arcs → …`

## Design Conformance

`architect-reviewer` sign-off: **7/7 CONFORMANT, 0 MISSING, 0 DEAD** against `docs/design/procedures/grow.md` Phase 3 (intersection detection) and Phase 1b (interleave beats) requirements.

Checked requirements:
1. ✅ Phase 3 executes before Phase 1b (dependency chain enforces this structurally)
2. ✅ Phase 3 input: "No predecessor edges exist yet (interleave has not run)"
3. ✅ Phase 3 preconditions (beats, belongs_to, locations, entities) satisfied right after validate_dag
4. ✅ No-Conditional-Prerequisites Invariant: interleave-created edges cannot invalidate intersection proposals
5. ✅ Co-grouped beats skip predecessor edges (dict[str, set[str]] with isdisjoint check)
6. ✅ Execution order in design doc matches implementation
7. ✅ entity_arcs still runs after intersections (priority 8 > 1)

No fixture divergence: test fixtures mirror `apply_intersection_mark` output exactly.

## Test plan
- [x] `test_phase_order_intersections_before_interleave` — replaces old `test_phase_order_gaps_before_intersections` with new invariant
- [x] `test_skips_predecessor_between_same_intersection_beats` — new test for intersection-aware interleave
- [x] `test_global_registry_execution_order_matches_expected` updated
- [x] `test_phase_order_names` updated (15 phases, new order)
- [x] All 308 unit tests pass

## Deferred
- `interleave_cycle_skipped` → hard error: tracked in #1129. The `resolve_temporal_hints` phase in #1128 prevents temporal-hint-caused cycles upstream; remaining edge cases tracked there.
- `scene_types` fragile dependency ordering (priority-only, no explicit `depends_on=["interleave_beats"]`): pre-existing, tracked in a follow-up.

Closes #1124

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -310,6 +310,7 @@ qf inspect -p <project> --json # Machine-readable JSON output
 - Using "tests pass" as sole evidence that a removal/refactoring issue is complete
 - Epics larger than 10 issues (split into sequential milestones)
 - Claiming "SEED/GROW/POLISH ran successfully" based on exit code without verifying output against design docs
+- **Silent degradation of story structure constraints** — if the pipeline cannot satisfy a structural requirement (cross-dilemma ordering, intersection formation, DAG consistency), it MUST fail loudly. Silently skipping constraints and producing a weaker story is not acceptable. `interleave_cycle_skipped` warnings, all-intersections-rejected, and similar "we tried but gave up" outcomes are pipeline failures, not warnings.
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply CLAUDE.md:313]: Good catch. Out of scope here — tracked in #1129.

**Comment by pvliesdonk** [comment]
Addressed all review comments:
- Replied to Gemini's `interleave_cycle_skipped` finding — tracked in #1129 (out of scope for this PR; the `resolve_temporal_hints` phase in #1128 should prevent the case upstream)

Conformance review: 7/7 CONFORMANT, 0 MISSING/DEAD against `docs/design/procedures/grow.md` Phase 3 and Phase 1b requirements.

File: src/questfoundry/graph/grow_algorithms.py [hunk 2]
```diff
@@ -2483,6 +2483,13 @@ def interleave_cross_path_beats(graph: Graph) -> int:
 
     beat_set = set(beat_nodes.keys())
 
+    # Build intersection-group index (beat → group_id) to avoid creating
+    # predecessor edges between beats that are co-grouped in an intersection.
+    # Such edges would create circular prerequisites on shared beats (#1124).
+    beat_intersection_group: dict[str, str] = {}
+    for edge in graph.get_edges(from_id=None, to_id=None, edge_type="intersection"):
+        beat_intersection_group[edge["from"]] = edge["to"]
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/questfoundry/graph/grow_algorithms.py]: Fixed in `eff4d88` — changed to `dict[str, set[str]]` with `.isdisjoint()` check. Also handles the case where `to_beat` has no groups (`get(to_beat, set())`).

File: src/questfoundry/graph/grow_algorithms.py [hunk 3]
```diff
@@ -2510,6 +2517,17 @@ def _add_predecessor(from_beat: str, to_beat: str) -> bool:
...
+        # such beats co-occur in a single scene and have no ordering (#1124).
+        from_group = beat_intersection_group.get(from_beat)
+        if from_group is not None and from_group == beat_intersection_group.get(to_beat):
+            log.debug(
+                "interleave_skipped_same_intersection",
+                from_beat=from_beat,
+                to_beat=to_beat,
+                group=from_group,
+            )
+            return False
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/questfoundry/graph/grow_algorithms.py]: Fixed in `eff4d88` — same commit as the dict change above.

File: src/questfoundry/pipeline/stages/grow/llm_phases.py [hunk 4]
```diff
@@ -336,7 +342,7 @@ async def _phase_3_intersections(self, graph: Graph, model: BaseChatModel) -> Gr
             tokens_used=total_tokens,
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/questfoundry/pipeline/stages/grow/llm_phases.py:342]: Acknowledged as pre-existing fragility. `scene_types` doesn't semantically depend on interleaved state — it only needs beats to exist (which validate_dag guarantees). The priority ordering is load-bearing but documented in the execution order note. Tracked as a follow-up to add explicit `depends_on=["interleave_beats"]` if the design ever requires it.

**Comment by pvliesdonk** [comment]
Addressed all remaining review comments (commit eff4d88):
- Fixed `beat_intersection_group` dict → `dict[str, set[str]]` with `.isdisjoint()` for correct multi-group membership handling
- Fixed stale `_phase_order()` docstring in `stage.py`
- Added `## Design Conformance` section to PR description (7/7 CONFORMANT)
- `interleave_cycle_skipped` → hard error: tracked in #1129; #1128's `resolve_temporal_hints` phase eliminates the temporal-hint-caused cases upstream
- `scene_types` priority fragility: acknowledged as pre-existing, no semantic dependency on interleaved state

/gemini review

File: src/questfoundry/graph/grow_algorithms.py [hunk 5]
```diff
@@ -2510,6 +2519,18 @@ def _add_predecessor(from_beat: str, to_beat: str) -> bool:
...
+        from_groups = beat_intersection_groups.get(from_beat, set())
+        if from_groups and not from_groups.isdisjoint(beat_intersection_groups.get(to_beat, set())):
+            shared = from_groups & beat_intersection_groups.get(to_beat, set())
+            log.debug(
+                "interleave_skipped_same_intersection",
+                from_beat=from_beat,
+                to_beat=to_beat,
+                groups=sorted(shared),
+            )
+            return False
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/questfoundry/graph/grow_algorithms.py:2534]: Applied in `225383a` — cleaner, avoids the double lookup.

File: src/questfoundry/pipeline/stages/grow/llm_phases.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/questfoundry/pipeline/stages/grow/llm_phases.py:342]: Fixed in `270e707` — changed `depends_on=["validate_dag"]` to `depends_on=["interleave_beats"]`. The ordering is now structural, not a priority tiebreak.

File: src/questfoundry/graph/grow_algorithms.py [hunk 6]
```diff
@@ -2483,6 +2483,15 @@ def interleave_cross_path_beats(graph: Graph) -> int:
...
     beat_set = set(beat_nodes.keys())
 
+    # Build intersection-group index (beat → set of group_ids) to avoid creating
+    # predecessor edges between beats that are co-grouped in an intersection.
+    # Such edges would create circular prerequisites on shared beats (#1124).
+    # A beat can theoretically belong to multiple intersection groups (e.g., grouped
+    # by both location and entity), so we track all groups per beat.
+    beat_intersection_groups: dict[str, set[str]] = {}
+    for edge in graph.get_edges(from_id=None, to_id=None, edge_type="intersection"):
+        beat_intersection_groups.setdefault(edge["from"], set()).add(edge["to"])
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/questfoundry/graph/grow_algorithms.py]: Applied in `a6222e2`.

File: tests/unit/test_grow_algorithms.py [hunk 7]
```diff
@@ -4594,3 +4597,42 @@ def test_temporal_hint_same_dilemma_skipped(self) -> None:
...
+
+    def test_skips_predecessor_between_same_intersection_beats(self) -> None:
+        """Beats co-grouped in an intersection must not get predecessor edges (#1124).
+
+        When two beats are in the same intersection group they co-occur in a
+        single scene — ordering them relative to each other is meaningless and
+        would create circular prerequisites if they also have cross-dilemma
+        relationship edges.
+        """
+        graph = _make_two_dilemma_graph_with_relationship("concurrent")
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply tests/unit/test_grow_algorithms.py]: Good catch — fixed in `babbbd5`. Changed to `"wraps"` which attempts `predecessor(aq_intro → mt_intro)` between the intro beats, so the skip logic is actually exercised.

**Comment by pvliesdonk** [comment]
Closing and re-opening as a clean PR with all fixes incorporated from the start.
